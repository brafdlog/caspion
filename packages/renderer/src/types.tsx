/// ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
// Types common with the electron code

import { type Auth, type drive_v3 } from 'googleapis';
import { type Transaction } from 'israeli-bank-scrapers-core/lib/transactions';
import { type BudgetSummary, type Account as YnabAccount } from 'ynab';

export type Credentials = Auth.Credentials;
export type Spreadsheet = Pick<drive_v3.Schema$File, 'id' | 'name'>;

/// ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
export enum OutputVendorName {
  YNAB = 'ynab',
  GOOGLE_SHEETS = 'googleSheets',
  JSON = 'json',
  CSV = 'csv',
}

export interface Config {
  outputVendors: {
    [OutputVendorName.GOOGLE_SHEETS]?: GoogleSheetsConfig;
    [OutputVendorName.YNAB]?: YnabConfig;
    [OutputVendorName.JSON]?: JsonConfig;
    [OutputVendorName.CSV]?: CsvConfig;
  };
  scraping: {
    timeout: number;
    numDaysBack: number;
    showBrowser: boolean;
    accountsToScrape: AccountToScrapeConfig[];
    chromiumPath?: string;
    maxConcurrency?: number;
    periodicScrapingIntervalHours?: number;
  };
}

export interface AppInfo {
  sourceCommitShort: string;
  repository: string;
  discordChanel: string;
  currentVersion: string;
}

export type OutputVendorConfigs = Exclude<Config['outputVendors'][OutputVendorName], undefined>;
export type OutputVendorConfig<T extends OutputVendorName> = Exclude<Config['outputVendors'][T], undefined>;

export interface OutputVendorConfigBase {
  active: boolean;
}

export interface CsvConfig extends OutputVendorConfigBase {
  options: {
    filePath: string;
  };
}

export interface JsonConfig extends OutputVendorConfigBase {
  options: {
    filePath: string;
  };
}

export interface GoogleSheetsConfig extends OutputVendorConfigBase {
  options: {
    credentials: Auth.Credentials;
    spreadsheetId: string;
  };
}

export interface YnabConfig extends OutputVendorConfigBase {
  options: {
    accessToken: string;
    accountNumbersToYnabAccountIds: Record<string, { ynabBudgetId: string; ynabAccountId: string }>;
    budgetId: string;
    maxPayeeNameLength?: number;
  };
}

export interface AccountToScrapeConfig {
  id: string;
  key: CompanyTypes;
  name: string;
  loginFields: Record<string, string>;
  active?: boolean;
}

export enum CompanyTypes {
  HAPOALIM = 'hapoalim',
  HAPOALIM_BE_ONLINE = 'hapoalimBeOnline',
  BEINLEUMI = 'beinleumi',
  UNION = 'union',
  AMEX = 'amex',
  ISRACARD = 'isracard',
  VISACAL = 'visaCal',
  MAX = 'max',
  LEUMI_CARD = 'leumiCard',
  OTSAR_HAHAYAL = 'otsarHahayal',
  DISCOUNT = 'discount',
  MIZRAHI = 'mizrahi',
  LEUMI = 'leumi',
  MASSAD = 'massad',
  YAHAV = 'yahav',
  BEYAHAD_BISHVILHA = 'beyahadBishvilha',
  BEHATSDAA = 'behatsdaa',
  MERCANTILE = 'mercantile',
}

export enum AccountType {
  IMPORTER = 'Importer',
  EXPORTER = 'Exporter',
}

export enum AccountStatus {
  IDLE = 'idle',
  PENDING = 'pending',
  IN_PROGRESS = 'inProgress',
  DONE = 'done',
  ERROR = 'error',
}

export class BudgetTrackingEvent {
  message: string;

  vendorId?: CompanyTypes | OutputVendorName;

  accountStatus?: AccountStatus;

  error?: Error;

  accountType?: AccountType;

  constructor({
    message,
    vendorId,
    error,
    accountType,
    accountStatus = AccountStatus.IN_PROGRESS,
  }: BudgetTrackingEvent) {
    this.message = message;
    this.vendorId = vendorId;
    this.error = error;
    this.accountType = accountType;
    this.accountStatus = accountStatus;
  }
}

export class DownloadChromeEvent extends BudgetTrackingEvent {
  percent: number;

  constructor(percent: number) {
    super({ message: `Download Chrome: ${percent}%` });
    this.percent = percent;
  }
}

export interface EnrichedTransaction extends Transaction {
  accountNumber: string;
  category?: string;
  hash: string;
}

export interface ExporterEvent extends BudgetTrackingEvent {
  allTransactions: EnrichedTransaction[];
}

export interface ExporterEndEvent extends ExporterEvent {
  exportedTransactionsNum: number;
}

/// ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
// End of common types
/// ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

export interface Log {
  message: string;
  originalEvent?: BudgetTrackingEvent;
}

export interface Account {
  id: string;
  companyId: CompanyTypes | OutputVendorName;
  displayName: string;
  logo: string;
  type: AccountType;
  active: boolean;
  status?: AccountStatus;
  logs: Log[];
}

export interface Importer extends Account {
  loginFields: Record<string, string>;
}

export interface Exporter extends Account {
  options: object;
}

export interface AccountMetadata {
  companyId: string;
  companyName: string;
  logo: string;
}

export enum ModalStatus {
  IMPORTER_SETTINGS,
  EXPORTER_SETTINGS,
  LOGS,
  HIDDEN,
  NEW_SCRAPER,
  GENERAL_SETTINGS,
}

export enum ExporterResultType {
  WEBSITE_URL,
  FILE,
}

export interface ExportResultMetadata {
  resultType: ExporterResultType;
  getResultUri(exporter: OutputVendorConfigBase): string;
}
export type YnabFinancialAccount = Pick<YnabAccount, 'id' | 'name' | 'type'> & {
  budgetId: string;
  active: boolean;
};
export interface YnabAccountDetails {
  budgets: BudgetSummary[];
  accounts: YnabFinancialAccount[];
  categories?: string[];
}
export interface FinancialAccountDetails {
  name: string;
  accountNumber: string;
}
export enum FETCH_YNAB_ACCOUNT_DATA_STATUS {
  SUCCESS = 'SUCCESS',
  INVALID_ACCESS_TOKEN = 'INVALID_ACCESS_TOKEN',
  INVALID_BUDGET_ID = 'INVALID_BUDGET_ID',
  GENERAL_ERROR = 'GENERAL_ERROR',
}
export interface YnabAccountDataType {
  ynabAccountData?: YnabAccountDetails;
  financialAccountDetails?: FinancialAccountDetails[];
  status: FETCH_YNAB_ACCOUNT_DATA_STATUS;
}
