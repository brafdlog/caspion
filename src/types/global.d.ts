import ElectronLog from "electron-log";

declare global {
  const SENTRY_DSN: string
  const GOOGLE_CLIENT_ID: string
  const GOOGLE_CLIENT_SECRET: string
  const SEGMENT_WRITE_KEY: string
  const SOURCE_COMMIT_SHORT: string
  const APP_NAME: string
  const DISCORD_CHANNEL: string
  namespace NodeJS {
    interface Global {
      logger: ElectronLog.ElectronLog & {
        default: ElectronLog.ElectronLog;
      }
    }
  }
}
