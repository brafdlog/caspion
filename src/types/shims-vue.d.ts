declare module "*.vue" {
  import Vue from "vue";
  export default Vue;
}

declare module '*.png' {}
declare module '*.jpeg' {}
declare module '*.jpg' {}
