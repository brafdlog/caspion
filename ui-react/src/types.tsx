/// ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
// Types common with the electron code
/// ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
export enum OutputVendorName {
  YNAB = 'ynab',
  GOOGLE_SHEETS = 'googleSheets',
  JSON = 'json',
  CSV = 'csv'
}

export interface Config {
  outputVendors: {
    [OutputVendorName.GOOGLE_SHEETS]?: GoogleSheetsConfig;
    [OutputVendorName.YNAB]?: YnabConfig;
    [OutputVendorName.JSON]?: JsonConfig;
    [OutputVendorName.CSV]?: CsvConfig;
  };
  scraping: {
    numDaysBack: number;
    showBrowser: boolean;
    accountsToScrape: AccountToScrapeConfig[];
    chromiumPath?: string;
    maxConcurrency?: number;
  };
}

export type OutputVendorConfigs = Exclude<Config['outputVendors'][OutputVendorName], undefined>
export type OutputVendorConfig<T extends OutputVendorName> = Exclude<Config['outputVendors'][T], undefined>

interface OutputVendorConfigBase {
  active: boolean;
}

export interface CsvConfig extends OutputVendorConfigBase {
  options: {
    filePath: string;
  }
}

export interface JsonConfig extends OutputVendorConfigBase {
  options: {
    filePath: string;
  }
}

export interface GoogleSheetsConfig extends OutputVendorConfigBase {
  options: {
    // credentials: Credentials;
    credentials: string;
    spreadsheetId: string;
  }
}

export interface YnabConfig extends OutputVendorConfigBase {
  options: {
    accessToken: string;
    accountNumbersToYnabAccountIds: { [key: string]: string };
    budgetId: string;
    maxPayeeNameLength?: number;
  };
}

export interface AccountToScrapeConfig {
  id: string;
  // key: CompanyTypes;
  key: string;
  name: string;
  loginFields: Record<string, string>;
  active?: boolean;
}

export enum CompanyTypes {
  hapoalim = 'hapoalim',
  hapoalimBeOnline = 'hapoalimBeOnline',
  beinleumi = 'beinleumi',
  union = 'union',
  amex = 'amex',
  isracard = 'isracard',
  visaCal = 'visaCal',
  max = 'max',
  leumiCard = 'leumiCard',
  otsarHahayal = 'otsarHahayal',
  discount = 'discount',
  mizrahi = 'mizrahi',
  leumi = 'leumi',
  massad = 'massad'
}

export enum AccountType {
  IMPORTER = 'Importer', EXPORTER = 'Exporter'
}

export enum AccountStatus {
  IDLE = 'IDLE',
  PENDING = 'PENDING',
  IN_PROGRESS = 'IN_PROGRESS',
  DONE = 'DONE',
  ERROR = 'ERROR'
}

export type BudgetTrackingEvent = {
  message: string;

  vendorId?: CompanyTypes | OutputVendorName;

  accountStatus?: AccountStatus;

  error?: Error;

  accountType?: AccountType;
}

/// ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
// End of common types
/// ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

export interface Log {
  message: string;
  originalEvent?: BudgetTrackingEvent
}

export interface Account {
  id: string;
  companyId: string;
  displayName: string;
  logo: string;
  type: AccountType;
  active: boolean;
  status?: AccountStatus;
  logs: Log[]
}

export interface Importer extends Account {
  loginFields: Record<string, string>;
}

export interface Exporter extends Account {
  options: object
}

export type AccountMetadata = {
  companyId: string,
  companyName: string,
  logo: string
};

export enum ModalStatus {
  ImporterSettings,
  SettingsExporter,
  Logs,
  Hidden,
  NewScraper,
  GeneralSettings
}
