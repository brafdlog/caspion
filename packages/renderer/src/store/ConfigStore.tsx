import { openExternal, openItem, updateConfig } from '#preload';
import { autorun, makeAutoObservable, toJS } from 'mobx';
import { createContext, useContext } from 'react';
import accountMetadata, { exporterUIHandlers } from '../accountMetadata';
import {
  AccountStatus,
  AccountType,
  ExporterResultType,
  type Account,
  type AccountToScrapeConfig,
  type BudgetTrackingEvent,
  type CompanyTypes,
  type Config,
  type Exporter,
  type Importer,
  type Log,
  type OutputVendorName,
} from '../types';

interface AccountScrapingData {
  logs: Log[];
  status: AccountStatus;
}

const createAccountToScrapeConfigFromImporter = (
  importerConfig: Importer,
): AccountToScrapeConfig => ({
  id: importerConfig.id,
  active: importerConfig.active,
  key: importerConfig.companyId as CompanyTypes,
  loginFields: importerConfig.loginFields,
  name: importerConfig.displayName,
});

const createOutputVendorConfigFromExporter = (exporterConfig: Exporter) => ({
  active: exporterConfig.active,
  options: exporterConfig.options,
});

const createAccountObject = (
  id: string,
  companyId: keyof typeof accountMetadata,
  type: AccountType,
  active: boolean,
  accountScrapingData?: AccountScrapingData,
): Account => {
  const metadata = accountMetadata[companyId];
  if (!metadata) {
    throw new Error(`No metadata found for companyId ${companyId}`);
  }
  return {
    id,
    companyId,
    displayName: metadata.companyName,
    logo: metadata.logo,
    type,
    active,
    status: accountScrapingData?.status ?? AccountStatus.IDLE,
    logs: accountScrapingData?.logs ?? [],
  };
};

const saveConfigIntoFile = (config?: Config) => {
  if (!config) {
    console.warn(`Can't save config into file. Config is ${config}`);
    return;
  }
  updateConfig(toJS(config));
};

class ConfigStore {
  config: Config;

  chromeDownloadPercent = 0;

  // TODO: move this to a separate store
  accountScrapingData: Map<CompanyTypes | OutputVendorName, AccountScrapingData>;

  constructor() {
    this.accountScrapingData = new Map();
    makeAutoObservable(this);

    autorun(() => {
      saveConfigIntoFile(this.config);
    });
  }

  updateConfig(config: Config) {
    this.config = config;
  }

  get importers(): Importer[] {
    if (!this.config) return [];
    return this.config.scraping.accountsToScrape.map(({ id, key, active, loginFields }) => {
      return {
        ...createAccountObject(
          id,
          key,
          AccountType.IMPORTER,
          !!active,
          this.accountScrapingData.get(key),
        ),
        loginFields,
      };
    });
  }

  get exporters(): Exporter[] {
    if (!this.config) return [];
    return Object.entries(this.config.outputVendors).map(([exporterKey, exporter]) => {
      return {
        ...createAccountObject(
          exporterKey,
          exporterKey as OutputVendorName,
          AccountType.EXPORTER,
          !!exporter?.active,
          this.accountScrapingData.get(exporterKey as OutputVendorName),
        ),
        options: exporter?.options || {},
      };
    });
  }

  get isScraping(): boolean {
    return !!Array.from(this.accountScrapingData.values()).find(
      account => account.status === AccountStatus.IN_PROGRESS,
    );
  }

  get settings() {
    return {
      numDaysBack: this.config?.scraping.numDaysBack,
      showBrowser: this.config?.scraping.showBrowser,
    };
  }

  clearScrapingStatus() {
    this.accountScrapingData = new Map();
    this.updateChromeDownloadPercent(0);
  }

  updateChromeDownloadPercent(percent: number) {
    this.chromeDownloadPercent = percent;
  }

  // TODO: not need to be in a store
  openResults(exporterName: OutputVendorName) {
    const exporterConfig = this.config.outputVendors[exporterName];
    if (!exporterConfig) {
      throw new Error(`No exporter found for ${exporterName}`);
    }
    const { resultType, getResultUri } = exporterUIHandlers[exporterName];
    const uri = getResultUri(exporterConfig);
    if (resultType === ExporterResultType.WEBSITE_URL) {
      openExternal(uri);
    } else {
      openItem(uri);
    }
  }

  handleScrapingEvent(eventName: string, budgetTrackingEvent?: BudgetTrackingEvent) {
    console.log('Received scraping event', eventName, budgetTrackingEvent);
    if (eventName === 'DOWNLOAD_CHROME') {
      this.updateChromeDownloadPercent((budgetTrackingEvent as DownalodChromeEvent)?.percent);
    }
    if (budgetTrackingEvent) {
      const accountId = budgetTrackingEvent.vendorId;
      if (accountId) {
        if (!this.accountScrapingData.has(accountId)) {
          this.accountScrapingData.set(accountId, {
            logs: [],
            status: AccountStatus.IDLE,
          });
        }
        const accountScrapingData = this.accountScrapingData.get(accountId);
        if (accountScrapingData) {
          accountScrapingData.logs.push({
            message: budgetTrackingEvent.message,
            originalEvent: budgetTrackingEvent,
          });
          accountScrapingData.status = budgetTrackingEvent.accountStatus ?? AccountStatus.IDLE;
        }
      }
    }
  }

  async addImporter(importerConfig: Importer) {
    if (!accountMetadata[importerConfig.companyId]) {
      throw new Error(`Company id ${importerConfig.companyId} is not a valid company id`);
    }
    const accountToScrapeConfig: AccountToScrapeConfig =
      createAccountToScrapeConfigFromImporter(importerConfig);
    this.config.scraping.accountsToScrape.push(accountToScrapeConfig);
  }

  async updateImporter(id: string, updatedImporterConfig: Importer) {
    const importerIndex = this.config.scraping.accountsToScrape.findIndex(
      importer => importer.id === id,
    );
    if (importerIndex === -1) {
      throw new Error(`Cant update importer with id ${id}. No importer with that id found`);
    }
    this.config.scraping.accountsToScrape[importerIndex] =
      createAccountToScrapeConfigFromImporter(updatedImporterConfig);
  }

  async deleteImporter(id: string) {
    this.config.scraping.accountsToScrape = this.config.scraping.accountsToScrape.filter(
      importer => importer.id !== id,
    );
  }

  async updateExporter(updatedExporterConfig: Exporter) {
    this.config.outputVendors[updatedExporterConfig.companyId as OutputVendorName] =
      createOutputVendorConfigFromExporter(updatedExporterConfig);
  }

  async toggleShowBrowser() {
    this.config.scraping.showBrowser = !this.config.scraping.showBrowser;
  }

  async setNumDaysBack(numDaysBack: number) {
    this.config.scraping.numDaysBack = numDaysBack;
  }

  async setTimeout(timeout: number) {
    this.config.scraping.timeout = timeout;
  }

  async setMaxConcurrency(maxConcurrency: number) {
    this.config.scraping.maxConcurrency = maxConcurrency;
  }

  async setChromiumPath(chromiumPath?: string) {
    this.config.scraping.chromiumPath = chromiumPath;
  }
}

export const configStore = new ConfigStore();
const StoreContext = createContext<ConfigStore>(configStore);
export const ConfigStoreProvider = ({ children }: { children: React.ReactNode }) => (
  <StoreContext.Provider value={configStore}>{children}</StoreContext.Provider>
);
export const useConfigStore = () => useContext(StoreContext);
