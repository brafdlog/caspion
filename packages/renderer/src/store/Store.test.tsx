import { type ConfigStore, configStore } from './ConfigStore';
import { AccountType, CompanyTypes, type Config, type Credentials, type Exporter, type Importer } from '../types';
import { beforeEach, describe, test, expect } from 'vitest';

describe('Store', () => {
  let store: ConfigStore;
  beforeEach(() => {
    store = configStore;
    store.config = dummyConfig;
  });
  describe('Properties and getters', () => {
    test('basic', () => {
      const basicStore = configStore;

      basicStore.config = dummyConfig;

      expect(basicStore.config).toMatchObject(dummyConfig);

      expect(basicStore.settings).toMatchSnapshot();
    });

    test('importers', () => {
      expect(store.importers).toHaveLength(3);

      const discountImporter = store.importers.find((importer) => importer.companyId === 'discount');
      const visaCalImporter = store.importers.find((importer) => importer.companyId === 'visaCal');
      const maxImporter = store.importers.find((importer) => importer.companyId === 'max');

      expect(discountImporter).toMatchSnapshot();
      expect(visaCalImporter).toMatchSnapshot();
      expect(maxImporter).toMatchSnapshot();
    });
    test('exporters', () => {
      expect(store.exporters).toHaveLength(4);

      expect(store.exporters).toMatchSnapshot();
    });

    test('settings', () => {
      expect(store.settings.numDaysBack).toEqual(dummyConfig.scraping.numDaysBack);
      expect(store.settings.showBrowser).toEqual(dummyConfig.scraping.showBrowser);
    });
  });
  describe('actions', () => {
    describe('importers', () => {
      test('addImporter', () => {
        const accountToAdd: Importer = {
          id: 'moshe',
          companyId: CompanyTypes.HAPOALIM,
          displayName: 'פועלים',
          loginFields: {
            username: 'david',
          },
          logo: '',
          logs: [],
          type: AccountType.IMPORTER,
          active: false,
        };
        const originalImportersLength = store.importers.length;
        store.addImporter(accountToAdd);
        expect(store.importers).toHaveLength(originalImportersLength + 1);
      });

      test('updateImporter', () => {
        const importer = store.importers[0];
        const updatedImporter = {
          ...importer,
          active: !importer.active,
        };
        store.updateImporter(importer.id, updatedImporter);
        expect(store.importers.find((importer) => importer.id === importer.id)).toHaveProperty(
          'active',
          updatedImporter.active,
        );
      });

      test('deleteImporter', () => {
        const importer = dummyConfig.scraping.accountsToScrape[0];
        const originalNumImporters = dummyConfig.scraping.accountsToScrape.length;
        expect(store.importers).toHaveLength(originalNumImporters);
        store.deleteImporter(importer.id);
        expect(store.importers).toHaveLength(originalNumImporters - 1);
      });
    });
    describe('exporters', () => {
      test('updateExporter', () => {
        const exporter: Exporter = store.exporters[0];
        const updatedExporter: Exporter = {
          ...exporter,
          options: {
            moo: 'foo',
          },
        };
        store.updateExporter(updatedExporter);
        expect(store.exporters[0]).toMatchObject(updatedExporter);
      });
    });
  });
});

export const dummyConfig: Config = {
  scraping: {
    numDaysBack: 40,
    showBrowser: false,
    accountsToScrape: [
      {
        active: true,
        key: CompanyTypes.DISCOUNT,
        name: 'Discount Bank',
        loginFields: {
          id: '234234134',
          password: 'notARealPass',
          num: 'LY9999',
        },
        id: 'efsvxasgd416f94d1e45a9cfc8c0a95f',
      },
      {
        active: true,
        key: CompanyTypes.VISACAL,
        name: 'Visa Cal',
        loginFields: {
          username: 'someUsername',
          password: 'somePassword',
        },
        id: '794a9ksj532d86cb4xxxb3438d0da942',
      },
      {
        active: true,
        key: CompanyTypes.MAX,
        name: 'Max',
        loginFields: {
          username: 'whatAUserName',
          password: 'greatPass',
        },
        id: '9xve2485ac202f53f85bcdb4f795dssc',
      },
    ],
    timeout: 0,
  },
  outputVendors: {
    csv: {
      active: true,
      options: {
        filePath: 'transaction.csv',
      },
    },
    json: {
      active: false,
      options: {
        filePath: 'transaction.json',
      },
    },
    ynab: {
      active: true,
      options: {
        accessToken: '################-######-####-###',
        budgetId: 'advasdvasd-asdvasdva-sdvasdvasdv-asdvasdvf',
        accountNumbersToYnabAccountIds: {
          5555: { ynabBudgetId: 'advasdvasd-asdvasdva-sdvasdvasdv-asdvasdvf', ynabAccountId: 'asdvasdvsdvs' },
          1111: { ynabBudgetId: 'advasdvasd-asdvasdva-sdvasdvasdv-asdvasdvf', ynabAccountId: 'xcvxcvxcvsd' },
          3333: { ynabBudgetId: 'advasdvasd-asdvasdva-sdvasdvasdv-asdvasdvf', ynabAccountId: 'xvxcdv' },
          555555555: { ynabBudgetId: 'advasdvasd-asdvasdva-sdvasdvasdv-asdvasdvf', ynabAccountId: 'vsdvsdvserverv' },
        },
      },
    },
    googleSheets: {
      active: false,
      options: {
        spreadsheetId: '',
        credentials: {} as Credentials,
      },
    },
  },
};
