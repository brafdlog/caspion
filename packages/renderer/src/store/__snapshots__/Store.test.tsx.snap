// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Store > Properties and getters > basic 1`] = `
{
  "numDaysBack": 40,
  "showBrowser": false,
}
`;

exports[`Store > Properties and getters > exporters 1`] = `
[
  {
    "active": true,
    "companyId": "csv",
    "displayName": "אקסל",
    "id": "csv",
    "logo": "/src/assets/exporters/excel.png",
    "logs": [],
    "options": {
      "filePath": "transaction.csv",
    },
    "status": "idle",
    "type": "Exporter",
  },
  {
    "active": false,
    "companyId": "json",
    "displayName": "Json",
    "id": "json",
    "logo": "/src/assets/exporters/json.jpeg",
    "logs": [],
    "options": {
      "filePath": "transaction.json",
    },
    "status": "idle",
    "type": "Exporter",
  },
  {
    "active": true,
    "companyId": "ynab",
    "displayName": "Ynab",
    "id": "ynab",
    "logo": "/src/assets/exporters/ynab.jpeg",
    "logs": [],
    "options": {
      "accessToken": "################-######-####-###",
      "accountNumbersToYnabAccountIds": {
        "1111": {
          "ynabAccountId": "xcvxcvxcvsd",
          "ynabBudgetId": "advasdvasd-asdvasdva-sdvasdvasdv-asdvasdvf",
        },
        "3333": {
          "ynabAccountId": "xvxcdv",
          "ynabBudgetId": "advasdvasd-asdvasdva-sdvasdvasdv-asdvasdvf",
        },
        "5555": {
          "ynabAccountId": "asdvasdvsdvs",
          "ynabBudgetId": "advasdvasd-asdvasdva-sdvasdvasdv-asdvasdvf",
        },
        "555555555": {
          "ynabAccountId": "vsdvsdvserverv",
          "ynabBudgetId": "advasdvasd-asdvasdva-sdvasdvasdv-asdvasdvf",
        },
      },
      "budgetId": "advasdvasd-asdvasdva-sdvasdvasdv-asdvasdvf",
    },
    "status": "idle",
    "type": "Exporter",
  },
  {
    "active": false,
    "companyId": "googleSheets",
    "displayName": "Google Sheets",
    "id": "googleSheets",
    "logo": "/src/assets/exporters/sheets.png",
    "logs": [],
    "options": {
      "credentials": {},
      "spreadsheetId": "",
    },
    "status": "idle",
    "type": "Exporter",
  },
]
`;

exports[`Store > Properties and getters > importers 1`] = `
{
  "active": true,
  "companyId": "discount",
  "displayName": "דיסקונט",
  "id": "efsvxasgd416f94d1e45a9cfc8c0a95f",
  "loginFields": {
    "id": "234234134",
    "num": "LY9999",
    "password": "notARealPass",
  },
  "logo": "/src/assets/importers/discount.jpeg",
  "logs": [],
  "status": "idle",
  "type": "Importer",
}
`;

exports[`Store > Properties and getters > importers 2`] = `
{
  "active": true,
  "companyId": "visaCal",
  "displayName": "ויזה כאל",
  "id": "794a9ksj532d86cb4xxxb3438d0da942",
  "loginFields": {
    "password": "somePassword",
    "username": "someUsername",
  },
  "logo": "/src/assets/importers/visa-cal.jpeg",
  "logs": [],
  "status": "idle",
  "type": "Importer",
}
`;

exports[`Store > Properties and getters > importers 3`] = `
{
  "active": true,
  "companyId": "max",
  "displayName": "Max",
  "id": "9xve2485ac202f53f85bcdb4f795dssc",
  "loginFields": {
    "password": "greatPass",
    "username": "whatAUserName",
  },
  "logo": "/src/assets/importers/max.jpg",
  "logs": [],
  "status": "idle",
  "type": "Importer",
}
`;
